cli/cmd/repair.go

```diff
@@ -0,0 +1,124 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	pb "github.com/linkerd/linkerd2/controller/gen/config"
+	"github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// newCmdRepair creates a new cobra command `repair` which re-creates the
+// linkerd-config-overrides secret if it has been deleted.
+func newCmdRepair() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "repair",
+		Short: "Output the secret/linkerd-config-overrides resource if it has been deleted",
+		Long: `Output the secret/linkerd-config-overrides resource if it has been deleted.
+
+The secret/linkerd-config-overrides resource is necessary to perform upgrades of
+the Linkerd control plane using the linkerd upgrade command. If this resource
+has been deleted, the linkerd repair command can make a best effort to restore
+it.  It is recommended that you review the secret/linkerd-config-overrides
+resource after running linkerd repair to avoid any unexpected behavior during
```

Comment:
I still don't understand how anyone is supposed to do review the resource. Should we have a `linkerd repair --plaintext` or something that actually shows the structure?

Reply:
Perhaps `linkerd repair --review` and `linkerd repair --secret`? With the default behavior (if no flag is specified) outputing help?

Ideally, the long help would include examples

Reply:
per @alpeb's suggestion, I am updating this to use the StringData field to make the output human readable:

```
bin/linkerd repair
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    linkerd.io/control-plane-ns: linkerd
  name: linkerd-config-overrides
  namespace: linkerd
stringData:
  linkerd-config-overrides: |
    global:
      identityTrustAnchorsPEM: |
        -----BEGIN CERTIFICATE-----
        MIIBgzCCASmgAwIBAgIBATAKBggqhkjOPQQDAjApMScwJQYDVQQDEx5pZGVudGl0
        eS5saW5rZXJkLmNsdXN0ZXIubG9jYWwwHhcNMjEwMjA0MDAxODU2WhcNMjIwMjA0
        MDAxOTE2WjApMScwJQYDVQQDEx5pZGVudGl0eS5saW5rZXJkLmNsdXN0ZXIubG9j
        YWwwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASZjqiX51ftr+n7QO061UffKzdN
        BtDFw1UO5lCb7CAM4eSF379WkgoSpy2wNZ6gacx+IHgK8lMYN8zGipK15DfPo0Iw
        QDAOBgNVHQ8BAf8EBAMCAQYwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMC
        MA8GA1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0EAwIDSAAwRQIhAKZz0HUumz479RDI
        w6TzTqOTYbMGLRUllBoPg05x51ZhAiAb4zDUAWCviZv/P8RkDS0BYSIvLJ21cT5g
        M13fSmbzfg==
        -----END CERTIFICATE-----
    identity:
      issuer:
        crtExpiry: "2022-02-04T00:19:16Z"
        tls:
          crtPEM: |-
            -----BEGIN CERTIFICATE-----
            MIIBgzCCASmgAwIBAgIBATAKBggqhkjOPQQDAjApMScwJQYDVQQDEx5pZGVudGl0
            eS5saW5rZXJkLmNsdXN0ZXIubG9jYWwwHhcNMjEwMjA0MDAxODU2WhcNMjIwMjA0
            MDAxOTE2WjApMScwJQYDVQQDEx5pZGVudGl0eS5saW5rZXJkLmNsdXN0ZXIubG9j
            YWwwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASZjqiX51ftr+n7QO061UffKzdN
            BtDFw1UO5lCb7CAM4eSF379WkgoSpy2wNZ6gacx+IHgK8lMYN8zGipK15DfPo0Iw
            QDAOBgNVHQ8BAf8EBAMCAQYwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMC
            MA8GA1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0EAwIDSAAwRQIhAKZz0HUumz479RDI
            w6TzTqOTYbMGLRUllBoPg05x51ZhAiAb4zDUAWCviZv/P8RkDS0BYSIvLJ21cT5g
            M13fSmbzfg==
            -----END CERTIFICATE-----
          keyPEM: |-
            -----BEGIN EC PRIVATE KEY-----
            MHcCAQEEIG19CmVq+Pxq0PbS4WQj4gq1eZ8ZLhugtjS0kuphoyhjoAoGCCqGSM49
            AwEHoUQDQgAEmY6ol+dX7a/p+0DtOtVH3ys3TQbQxcNVDuZQm+wgDOHkhd+/VpIK
            EqctsDWeoGnMfiB4CvJTGDfMxoqSteQ3zw==
            -----END EC PRIVATE KEY-----
```